File: FactoryDesignPattern/Shapes_Example/Driver.java
package FactoryDesignPattern.Shapes_Example;

public class Driver {
  public static void main(String[] args) {
    ShapeFactory circleFactory = new CircleFactory();
    ShapeFactory rectangleFactory = new RectangleFactory();
    ShapeFactory triangleFactory = new TriangleFactory();

    Client user1 = new Client(circleFactory);
    Client user2 = new Client(rectangleFactory);
    Client user3 = new Client(triangleFactory);

    Shape createdShapebyUser1 = user1.getShape();
    Shape createdShapebyUser2 = user2.getShape();
    Shape createdShapebyUser3 = user3.getShape();

    createdShapebyUser1.describe();
    createdShapebyUser2.describe();
    createdShapebyUser3.describe();
  }
}

File: FactoryDesignPattern/Shapes_Example_Python/Driver.py
from Client import Client
from Shapes.Shape import Shape
from Shapes_Factory.CrircleFactory import CircleFactory
from Shapes_Factory.RectangleFactory import RectangleFactory
from Shapes_Factory.TriangleFactory import TriangleFactory

class Driver():

  @staticmethod
  def main():
    client1: Client = Client(CircleFactory())
    client2: Client = Client(RectangleFactory())
    client3: Client = Client(TriangleFactory())

    shape1CreatedClient1: Shape = client1.getShape()
    shape2CreatedClient2: Shape = client2.getShape()
    shape3CreatedClient3: Shape = client3.getShape()

    shape1CreatedClient1.describe()
    shape2CreatedClient2.describe()
    shape3CreatedClient3.describe()

if __name__ == "__main__":
  Driver.main()